### ISSUE TITLE
UnnecessaryParentheses reports incorrect violation message for return value in lambdas

### ISSUE THREAD

**Issue by faraazb** [checkstyle/checkstyle#11357]
I have read check documentation: https://checkstyle.org/config_coding.html#UnnecessaryParentheses
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```shell
$ javac ParenthesesTest.java
# ParenthesesTest.class created

$ type config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name = "Checker">
        <module name="TreeWalker">
                <module name="UnnecessaryParentheses"/>
        </module>
</module>

$ type ParenthesesTest.java
import java.util.function.Function;

public class ParenthesesTest {
    public void returnValueExample() {
        Function<Integer, Integer> addFive = x -> {
            return (x + 5); // violation, should be reported for parentheses around return value
        };
    }
}

$ java "-Duser.language=en -Duser.country=US" -jar checkstyle-10.0-all.jar -c config.xml ParenthesesTest.java
Starting audit...
[ERROR] C:\Users\Faraaz\CheckstyleTest\ParenthesesTest.java:6:20: Unnecessary parentheses around assignment right-hand side. [UnnecessaryParentheses]
Audit done.
Checkstyle ends with 1 errors.
```
---

When a lambda expression is assigned to a variable and has a return value surrounded in parentheses, the violation reported is ' Unnecessary parentheses around assignment right-hand side.'.

**Expected:**
The correct violation message should be 'Unnecessary parentheses around return value.'.
ParenthesesTest.java:6:20 should be reported for unnecessary parentheses around return value.

---

Could it be due to [this](https://github.com/checkstyle/checkstyle/blob/4981aad0c6a4270f618fcab1b50a00c0c22a33f0/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/UnnecessaryParenthesesCheck.java#L585), wherein the assignment depth is being checked **before** checking whether the parent token is a 'return'?

**Comment by romani** [comment]
Please report issue as defined (strictly follow) in issue template

**Comment by faraazb** [comment]
Updated as per issue template.

**Comment by romani** [comment]
Please remove comment, and remove unnecessary code to make easy to see miminal code and distinct violations.

**Comment by faraazb** [comment]
Removed header comment and omitted some code.

**Comment by romani** [comment]
please make code compilable, you missed "javac".
users/maintainer/contributor should copy output of "type" to their file and reproduce a problem.

**Comment by faraazb** [comment]
I edited so that only the file header comment is removed. After the `javac` command, do I also need to mention that it compiles successfully?

**Comment by romani** [comment]
We usually print that execution of javac return no output, as definition of compilable.

>InputUnnecessaryParenthesesCheckSwitchExpression.java:21:31

Please update output.

-------

Try to put yourself in foot of person who need to fix this issue. Any mismatched or missed details are very discouraging. Idea of this output is to let anybody in world copy results of type/cat to files, run the same command (copy-paste) and get the same result.

If something is not matching, frustration is growing and nobody wants to deal with issue that is not clear and not easy to reproduce.

**Comment by faraazb** [comment]
Updated the Checkstyle output. Also added indication after `javac` command to confirm compilation. I completely agree with you, I will take care next time. Sorry for the noise it has created in the comments.

**Comment by romani** [comment]
>InputUnnecessaryParenthesesCheckSwitchExpression.java:58:18

I still see line 58 in description, please copy paste output. As you play with trailing comment s to show where is violation, please make sure CLI output and comments matches.

**Comment by faraazb** [comment]
😅I was making a dumb mistake. Should be good now. Maybe I should have used a different and smaller code example.

**Comment by romani** [comment]
>Maybe I should have used a different and smaller code example.

Yes, please do

**Comment by OttoKaaij** [comment]
On this
